HIP/hip-991.md

```diff
@@ -203,35 +204,43 @@ message ConsensusUpdateTopicTransactionBody {
   /*
    * The custom fee to be assessed during a message submission to this topic. Null if the fees should not be updated.
    */
-  ConsensusCustomFeeList custom_fees = 12;
+  FixedCustomFeeList custom_fees = 12;
 
 }
 
 message FeeExemptKeyList {
   repeated Key keys = 1;
 }
 
-message ConsensusCustomFeeList {
-  repeated ConsensusCustomFee fees = 1;
+message FixedCustomFeeList {
+  repeated FixedCustomFee fees = 1;
 }
 ```
 
-#### ConsensusSubmitMessageTransactionBody
+#### TransactionBody
 
-The `ConsensusSubmitMessageTransactionBody` message is updated to include the optional `max_custom_fees` property for specifying the maximum fee that the user is willing to pay for the message.
+The `TransactionBody` message is updated to include the optional `max_custom_fees` property for specifying the maximum custom fee that the user is willing to pay for the transaction.
 
 ```protobuf
-message ConsensusSubmitMessageTransactionBody {
+message TransactionBody {
   [..]
   /**
-    * The maximum custom fee that the user is willing to pay for the message. This field will be ignored if `accept_all_custom_fees` is set to `true`.
+    * The maximum custom fee that the user is willing to pay for the message. If left empty, the user is willing to pay any custom fee.
+    * If used with a transaction type that does not support custom fee limits, the transaction will fail.
     */
-  repeated FixedFee max_custom_fees = 4;
-  
+  repeated CustomFeeLimit max_custom_fees = 1001;
```

Comment:
Are we sure we really want to add this field to the generic `TransactionBody` instead of a common structure added to each impacted individual transaction bodies? This is the first time we're adding a new generic field to this message in 5+ years. This is also the opposite approach we settled on for block stream when discussing a field that applied to only two transaction types to not put it in the generic `TransactionResult` and to put it only in the specific transaction output.

Reply:
We had decided in the meeting that this would go onto `TransactionBody` (the high level transaction object, not the transaction body data)